src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -171,10 +240,14 @@ void RaftReplDev::on_push_data_received(intrusive< sisl::GenericRpcData >& rpc_d
     data_service()
         .async_write(r_cast< const char* >(data), push_req->data_size(), rreq->local_blkid)
         .thenValue([this, rreq](auto&& err) {
-            RD_REL_ASSERT(!err, "Error in writing data"); // TODO: Find a way to return error to the Listener
-            rreq->state.fetch_or(uint32_cast(repl_req_state_t::DATA_WRITTEN));
-            rreq->data_written_promise.setValue();
-            RD_LOG(INFO, "Data Channel: Data Write completed rreq=[{}]", rreq->to_compact_string());
+            if (err) {
+                RD_DBG_ASSERT(false, "Error in writing data");
+                handle_error(rreq, ReplServiceError::DRIVE_WRITE_ERROR);
```

Comment:
If there are any errors, we should free the blks allocated ?

Reply:
handle_error method does that.